Mathlib/RingTheory/Ideal/Norm/RelNorm.lean

```diff
@@ -34,44 +35,53 @@ namespace Ideal
 
 open Submodule
 
-variable (R : Type*) [CommRing R] {S : Type*} [CommRing S] [Algebra R S]
+attribute [local instance] FractionRing.liftAlgebra
 
-/-- `Ideal.spanNorm R (I : Ideal S)` is the ideal generated by mapping `Algebra.norm R` over `I`.
+variable (R S : Type*) [CommRing R] [IsDomain R] {S : Type*} [CommRing S] [IsDomain S]
+variable [IsIntegrallyClosed R] [IsIntegrallyClosed S] [Algebra R S] [Module.Finite R S]
+variable [NoZeroSMulDivisors R S] [Algebra.IsSeparable (FractionRing R) (FractionRing S)]
+
+/-- `Ideal.spanNorm R (I : Ideal S)` is the ideal generated by mapping `Algebra.intNorm R S`
+over `I`.
 
 See also `Ideal.relNorm`.
 -/
 def spanNorm (I : Ideal S) : Ideal R :=
-  Ideal.span (Algebra.norm R '' (I : Set S))
+  Ideal.span (Algebra.intNorm R S '' (I : Set S))
```

Comment:
This could be written as
```
  Ideal.map (Algebra.intNorm R S) I
```
(and then some simps below need to have `Ideal.map` added).